libkbfs/conflict_resolver.go

```diff
@@ -442,28 +442,39 @@ func (sp crSortedPaths) Swap(i, j int) {
 	sp[j], sp[i] = sp[i], sp[j]
 }
 
-func fileWithConflictingWrite(unmergedChains, mergedChains *crChains,
+func createdFileWithConflictingWrite(unmergedChains, mergedChains *crChains,
 	unmergedOriginal, mergedOriginal BlockPointer) bool {
 	mergedChain := mergedChains.byOriginal[mergedOriginal]
 	unmergedChain := unmergedChains.byOriginal[unmergedOriginal]
-	if mergedChain != nil && unmergedChain != nil {
-		mergedSync := false
-		unmergedSync := false
-		for _, op := range mergedChain.ops {
-			if _, ok := op.(*syncOp); ok {
-				mergedSync = true
-				break
-			}
-		}
-		for _, op := range unmergedChain.ops {
-			if _, ok := op.(*syncOp); ok {
-				unmergedSync = true
-				break
-			}
-		}
-		return mergedSync && unmergedSync
+	if mergedChain == nil || unmergedChain == nil {
+		return false
+	}
+
+	unmergedWriteRange := unmergedChain.getCollapsedWriteRange()
+	mergedWriteRange := mergedChain.getCollapsedWriteRange()
+	// Are they exactly equivalent?
+	if writeRangesEquivalent(unmergedWriteRange, mergedWriteRange) {
+		unmergedChain.removeSyncOps()
+		return false
 	}
-	return false
+
+	// If the unmerged one is just a truncate, we can safely ignore
+	// the unmerged chain.
+	if len(unmergedWriteRange) == 1 && unmergedWriteRange[0].isTruncate() &&
+		unmergedWriteRange[0].Off == 0 {
+		unmergedChain.removeSyncOps()
+		return false
+	}
+
+	// If the merged one is just a truncate, we can safely ignore
+	// the unmerged chain.
```

Comment:
Truncates are defined as a `WriteRange` where `Len == 0`.  So they don't really have "sizes" per se.  If you mean what happens if the unmerged branch truncates to a different place, or even does some writes -- that's ok.  Here's we're just ignoring the merged truncate to the beginning of the file.  This function is only called for _created_ files, so there's no danger that the merged truncate to 0 actually changed the data in the file.